src/bodywork/git.py

```diff
@@ -41,18 +42,23 @@ def download_project_code_from_repo(
     :raises RuntimeError: If Git is not available on the system or the
         Git repository cannot be accessed.
     """
+    log = bodywork_log_factory()
     try:
         run(['git', '--version'], check=True)
     except CalledProcessError:
         raise RuntimeError('git is not available')
-
-    if (get_connection_protocol(url) is ConnectionPrototcol.SSH
-            and get_remote_repo_host(url) is GitRepoHost.GITHUB):
-        setup_ssh_for_github()
-
+    try:
+        if (get_connection_protocol(url) is ConnectionPrototcol.SSH
+                and get_remote_repo_host(url) is GitRepoHost.GITHUB):
+            setup_ssh_for_github()
+        elif SSH_GITHUB_KEY_ENV_VAR not in os.environ:
+            log.warning('Not configured for use with private GitHub repos')
+    except Exception as e:
+        msg = f'git clone failed - Unable to setup SSH for Github: {e}'
```

Comment:
Technically speaking, the `git clone` hasn't failed. Maybe,

> `Unable to setup SSH for GitHub and you are trying to connect via SSH`

?

Reply:
👍